### PR TITLE
Adds basic tests for planar image detector wrapper

### PR DESCRIPTION
**Pull request by paullewis**

### CODE REVIEW

File: src/planar/planar-detector_test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  it('accepts objects with IDs', () => {
+    assert.doesNotThrow(() => {
+      detector.addDetectionWithId(100, new Uint8Array(1));
+    });
+  });
+
+  it('accepts objects without IDs', () => {
+    assert.doesNotThrow(() => {
+      detector.addDetection(new Uint8Array(1));
```
Comment (mmocny) [inline src/planar/planar-detector_test.ts]: Is this expected to return anything?

File: src/planar/planar-detector_test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    });
+  });
+
+  it('cancels object detection', () => {
+    detector.addDetectionWithId(100, new Uint8Array(1));
+    detector.cancelDetection(100);
```
Comment (mmocny) [inline src/planar/planar-detector_test.ts]: Presumably `detector.process` will act differently after add/cancel, but I guess we cannot test this with Mock detector?

File: src/planar/planar-detector_test.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  });
+
+  it('processes data while initializing', () => {
+    detector = new PlanarTargetDetector();
+    const value = detector.process(new ImageData(1, 1), 1);
+    assert.equal(value.size(), 0);
+    assert.deepEqual(value.get(0), { id: 0 });
+  });
+
+  it('processes data', () => {
```
Comment (mmocny) [inline src/planar/planar-detector_test.ts]: I don't understand how this test differs from above.  Is there something in `processPreRuns` which delays until detector finishes initializing?

File: src/planar/planar-detector_test.ts (hunk 1 above)
Comment (paullewis) [reply src/planar/planar-detector_test.ts]: No, the wasm doesn't currently, but I would like it to if only for correctness.

File: src/planar/planar-detector_test.ts (hunk 2 above)
Comment (paullewis) [reply src/planar/planar-detector_test.ts]: Right.

File: src/planar/planar-detector_test.ts (hunk 3 above)
Comment (paullewis) [reply src/planar/planar-detector_test.ts]: There's a "before the wasm has booted" and "after the wasm has booted", and I want to make sure that we get a sensible response from the wrapper if the wasm is still firing up.

**Pull request merged**